public_html/wp-content/plugins/wordcamp-speaker-feedback/includes/admin.php

```diff
@@ -478,6 +482,95 @@ function handle_bulk_edit_actions( $action, $nonce ) {
 	return $messages;
 }
 
+/**
+ * Add extra actions to the list of links displayed for each comment.
+ *
+ * @param array      $actions An array of comment actions.
+ * @param WP_Comment $comment Comment data object.
+ *
+ * @return array
+ */
+function feedback_extra_actions( $actions, $comment ) {
+	$comment_id = $comment->comment_ID;
+	$feedback = get_feedback_comment( $comment );
+	if ( is_null( $feedback ) ) {
+		return $actions;
+	}
+
+	$unapprove_url = add_query_arg(
+		array(
+			'bulk_edit[]' => $comment_id,
+			'action' => 'unmark-inappropriate',
+			'_wpnonce' => wp_create_nonce( "inappropriate-comment_{ $comment_id }" ),
+			'bulk_edit_nonce' => wp_create_nonce( 'bulk_edit_' . COMMENT_TYPE ),
+		),
+		get_subpage_url( get_post_type( $comment->comment_post_ID ) )
```

Comment:
💯 Good idea falling back to bulk edit. Here's an edge case that might be worth considering, although almost everyone will have JS enabled. If the user is on a view other than "All" (e.g. looking at just Approved or Inappropriate feedbacks), or they're on a subsequent page, or they've sorted the table by a non-default column, those choices are represented as additional parameters in the URL query string, and they will be lost when they click this link.

Reply:
Yeah, I guess that's unexpected. There's still an edge case because we're not re-calculating pages (so you can end up in a "no comments found" state), but that happens for bulk edit too… We can think about fixing that in another PR though.

Reply:
(I'll make the change to keep the user on the same page in this PR, though)

Reply:
Done in bea41e69a2534d29b69c9bf974423f265ec1cd59